### ISSUE TITLE
How to add a llm as reward model?

### ISSUE THREAD

**Issue by zyf8818** [volcengine/verl#1572]
I want to use a local llm as reward model, but verl only support rule base and hf seqclassification model. 
I have tried to use vllm bulid a offline server, however I do not kown how to add this server to verl.

**Comment by yantijin** [comment]
same here

**Comment by zyf8818** [comment]
@yantijin I want to use vllm for batch inference, API is two slow for generate hundreds of responses, do you consider that?

**Comment by llp1992** [comment]
same here

**Comment by sweartoyou** [comment]
same here

**Comment by wuxibin89** [comment]
You can define a custom `compute_score` in a python file, for example llm_score.py
```python
import requests

def compute_score(data_source, solution_str, ground_truth, extra_info):
    resp = requests.post("http://localhost:8000/score", json={
        "data_source": data_source,
        "solution_str": solution_str,
        "ground_truth": ground_truth,
        "extra_info": extra_info
    })
    return resp.json()["score"]
```
Then in training script, specify it by
```bash
python3 -m verl.trainer.main_ppo \
    ...
    custom_reward_function.path=/path/to/llm_score.py \
    custom_reward_function.name=compute_score \
    ...
```

**Comment by zyf8818** [comment]
> You can define a custom `compute_score` in a python file, for example llm_score.py
> 
> import requests
> 
> def compute_score(data_source, solution_str, ground_truth, extra_info):
>     resp = requests.post("http://localhost:8000/score", json={
>         "data_source": data_source,
>         "solution_str": solution_str,
>         "ground_truth": ground_truth,
>         "extra_info": extra_info
>     })
>     return resp.json()["score"]
> Then in training script, specify it by
> 
> python3 -m verl.trainer.main_ppo \
>     ...
>     custom_reward_function.path=/path/to/llm_score.py \
>     custom_reward_function.name=compute_score \
>     ...

多谢多谢！
使用vllm在另一张卡上部署一个API服务？没试过这种方法，调用效率如何？

**Comment by JoeyJyL** [comment]
same here

**Comment by Kirrito-k423** [comment]
定时清理检查结论：该 issue 最近一次由当前运行账号（Kirrito-k423）之外账号产生的公开回复或公开引用是 2025-11-06T16:19:07Z，来源为 comment（JoeyJyL）：https://github.com/verl-project/verl/issues/1572#issuecomment-3498161243。截至 2026-07-06，该时间已经超过 6 个月。

我已检查该 issue 的公开 comments 和 timeline（issue 页面：https://github.com/verl-project/verl/issues/1572；comments/timeline 对应 GitHub 公开 issue 记录）。根据本次定时清理规则，这类长时间没有新的非本人公开互动的 open issue 将先关闭。如果作者或社区成员仍能提供新的复现、证据或需求背景，请继续评论请求重开，或创建新的 issue 并引用这里的上下文。

---

Scheduled cleanup result: the latest public reply or public reference from an account other than the current runner (Kirrito-k423) was at 2025-11-06T16:19:07Z, from comment by JoeyJyL: https://github.com/verl-project/verl/issues/1572#issuecomment-3498161243. As of 2026-07-06, that is more than 6 months ago.

I checked the public comments and timeline for this issue (issue page: https://github.com/verl-project/verl/issues/1572; comments/timeline are part of the public GitHub issue record). Under the scheduled cleanup rule, open issues with no recent non-runner public interaction are closed. If the author or community can still provide a fresh reproduction, evidence, or requirement context, please comment to request reopening or open a new issue and link this context.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
